src/components/forms/FormCheckSelect/FormCheckSelect.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useCallback, useEffect, useState } from "react";
+import CheckBoxItem from "@app/components/ui/CheckBoxItem";
+import { Control, RegisterOptions, useController } from "react-hook-form";
+import { useStyles } from "./styles";
+import { ScrollView, Text, View } from "react-native";
+
+interface CheckOption {
+  displayText: string;
+  value: string;
+}
+
+interface FormCheckSelectProps {
+  name: string;
+  control: Control;
+  rules?: Exclude<
+    RegisterOptions,
+    "valueAsNumber" | "valueAsDate" | "setValueAs"
+  >;
+  label?: string;
+  selectedOptions?: string[];
+  options: CheckOption[];
+  errorMessage?: string;
+}
+
+const FormCheckSelect: React.FC<FormCheckSelectProps> = ({
+  name,
+  control,
+  rules,
+  label,
+  selectedOptions = [],
+  options,
+  errorMessage,
+}) => {
+  if (options.length === 0) {
+    return <></>; // return empty if no options
+  }
+
+  const styles = useStyles();
+
+  const [curOptions, setCurOptions] = useState<Record<string, boolean>>({});
```

Comment:
this variable name, curOptions is triggering inner spelling nazi 😂

Reply:
Lol not sure what else to call it